src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt

```diff
@@ -86,6 +87,23 @@ class PrisonApiClient(private val prisonApiWebClient: WebClient) {
       .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
       .block()
   }
+
+  /**
+   * This will only return appointments that have an end time. We need this so we can match it to booked appointments in
+   * BVLS should we have to.
+   */
+  fun getScheduledAppointments(prisonCode: String, date: LocalDate, locationId: Long) =
+    prisonApiWebClient.get()
+      .uri { uriBuilder: UriBuilder ->
+        uriBuilder
+          .path("/api/schedules/{prisonCode}/appointments")
+          .queryParam("date", date)
+          .queryParam("locationId", locationId)
+          .build(prisonCode)
+      }
+      .retrieve()
+      .bodyToMono(typeReference<List<ScheduledAppointment>>())
+      .block()?.filter { it.endTime != null } ?: emptyList()
 }
```

Comment:
Should we filter VLBs out of this list, as we're also getting these from BVLS DB directly? And, its the local data where we know which vlb ID to exclude for on the edit journey.

Reply:
in fact, maybe the filters for null empty times and VLBs should be done where this list is used? That would make it more generic, in case there are other uses of this endpoint?